### PR TITLE
Expose RetryAfter value in error

### PR DESCRIPTION
**Pull request by flimzy**
Related to, and useful along with #273

### CODE REVIEW

**Comment by Pineapple217** [comment]
IMO the Error type should contain time.Duration instead of time.Time. If the user wants time.Time they can calculate it them self (or add a method to Error to calculate it on the fly if you want it in the lib). Because if you want time.Duration you are just calculating back the value (doing double work). I know its nitpicky but something to think about. Either way I would be happy with this feature.

**Comment by flimzy** [comment]
> IMO the Error type should contain time.Duration instead of time.Time.

That was my initial thought as well. The problem with a `time.Duration` is that it looses its meaning virtually immediately. If you're doing any sort of batch processing, or anything else that might delay the handling of an error, then that value cannot be relied on any more.  By returning a `time.Time`, we aren't forcing the consumer of this library into any specific error handling behavior or pattern.

**Comment by Pineapple217** [comment]
Fair enough, both have their pros and cons but `time.Time` is probably better fitting for this lib when you put it like that. More intuitive and flexible.

File: spotify_test.go [hunk 1]
```diff
@@ -104,6 +107,52 @@ func TestNewReleasesRateLimitExceeded(t *testing.T) {
 	}
 }
 
+func TestRateLimitExceededReportsRetryAfter(t *testing.T) {
+	t.Parallel()
+	const retryAfter = 2
+
+	handlers := []http.HandlerFunc{
```
Comment (strideynet) [inline spotify_test.go:114]: I've actually never thought about using a slice over a set of a funcs like this in a test - but this is quite nice!

**Review by strideynet — APPROVED** [review]

**Pull request merged**